framework/include/materials/TwoMaterialPropertyInterface.h

```diff
@@ -19,6 +19,8 @@ class TwoMaterialPropertyInterface;
 template <>
 InputParameters validParams<TwoMaterialPropertyInterface>();
 
+#define adGetNeighborMaterialProperty this->template getNeighborMaterialProperty
+
```

Comment:
I would like to see a `getNeighborADMaterialProperty` method added to this class with a corresponding `adGetNeighborADMaterialProperty` macro. The code should look something like:
```c++
template <typename T>
const ADMaterialPropertyObject<T> &
TwoMaterialPropertyInterface::getNeighborADMaterialProperty(const std::string & name)
{
  // Check if the supplied parameter is a valid input parameter key
  std::string prop_name = deducePropertyName(name);

  // Check if it's just a constant
  const ADMaterialPropertyObject<T> * default_property = defaultADMaterialProperty<T>(prop_name);
  if (default_property)
    return *default_property;
  else
    return _neighbor_material_data->getADProperty<T>(prop_name);
}
```
I would also really like to see a test with material properties that depend on a coupled variable to make sure that our ADDG system will work with ADMaterials

Reply:
I'll do this myself